pkgs/development/libraries/freeimage/default.nix

```diff
@@ -1,37 +1,23 @@
-{ lib, stdenv, fetchurl, unzip, darwin }:
-
-# TODO: consider unvendoring various dependencies (libpng, libjpeg,
-# libwebp, zlib, ...)
+{ lib, stdenv, fetchsvn, unzip, darwin, libtiff
+, libpng, zlib, libwebp, libraw, openexr, openjpeg
+, libjpeg, jxrlib, pkgconfig, fetchpatch }:
 
+# This has been partially adapted from https://src.fedoraproject.org/rpms/freeimage
 stdenv.mkDerivation {
-  name = "freeimage-3.18.0";
+  pname = "freeimage";
+  version = "unstable-r1859-3.19.0";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/freeimage/FreeImage3180.zip";
-    sha256 = "1z9qwi9mlq69d5jipr3v2jika2g0kszqdzilggm99nls5xl7j4zl";
+  src = fetchsvn {
+    url = "https://svn.code.sf.net/p/freeimage/svn/";
+    rev = "r1859";
+    sha256 = "1d94935aqbkb994nqkw7m8xcynyz9rm6k7k59igrbjak8b63qpi6";
   };
+  sourceRoot = "svn-rr1859/FreeImage/trunk";
 
-  patches = lib.optional stdenv.isDarwin ./dylib.patch;
-
-  buildInputs = [ unzip ] ++ lib.optional stdenv.isDarwin darwin.cctools;
-
-  prePatch = if stdenv.isDarwin then ''
-    sed -e 's/$(shell xcrun -find clang)/clang/g' \
-        -e 's/$(shell xcrun -find clang++)/clang++/g' \
-        -e "s|PREFIX = /usr/local|PREFIX = $out|" \
-        -e 's|-Wl,-syslibroot $(MACOSX_SYSROOT)||g' \
-        -e 's|-isysroot $(MACOSX_SYSROOT)||g' \
-        -e 's|	install -d -m 755 -o root -g wheel $(INCDIR) $(INSTALLDIR)||' \
-        -e 's| -m 644 -o root -g wheel||g' \
-        -i ./Makefile.osx
-    # Fix LibJXR performance timers
-    sed 's|^SRCS = \(.*\)$|SRCS = \1 Source/LibJXR/image/sys/perfTimerANSI.c|' -i ./Makefile.srcs
-  '' else ''
-    sed -e s@/usr/@$out/@ \
-        -e 's@-o root -g root@@' \
-        -e 's@ldconfig@echo not running ldconfig@' \
-        -i Makefile.gnu Makefile.fip
-  '';
+  patches = ./unbundle.diff;
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ unzip libtiff libpng zlib libwebp libraw openexr openjpeg libjpeg jxrlib ] ++ lib.optional stdenv.isDarwin darwin.cctools;
```

Comment:
If `unzip` is only used at build time it should be in `nativeBuildInputs`. ALso `cctools` should be in `nativeBuildInputs`.